### DISCUSSION TITLE
How can I catch an error in the Model event?

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#13251] [Help]
### Package

Panel builder

### Package Version

v3.2.73

### How can we help you?

Hello everyone,

I have a Model that has a rules() method that returns an array of rules for validation.
This Model also has a static::saving() in its booted() method, which is responsible for applying the rules present in rules() before saving.

```
class Something extends Model
{
    use SoftDeletes;

    protected $table = "somethings";

    protected $guarded = [];

   protected static function booted(): void{
        
        parent::booted();

        static::saving(function(Model $model){
            $rules = $model->rules();

            $validator = Validator::make($model->toArray(), $rules);

            if ($validator->fails()) {
                throw new ValidationException($validator);
            }
        });

    }

    public static function rules(): array{
        return [
            'category' => ["required", Rule::enum(Category::class)],
        ];
    }

}
```

The problem is that I can't capture this error in Filament.

When a validation exception occurs, the "save" button doesn't display any error message.

Any ideas?

**Comment by leandrocfe** [comment]
Why are you using rules in the model instead of in the fields?

https://filamentphp.com/docs/3.x/forms/validation
https://filamentphp.com/docs/3.x/forms/validation#custom-rules

You could also use hooks if you want to add a custom rule.
https://filamentphp.com/docs/3.x/panels/resources/creating-records#lifecycle-hooks

**Reply by dan-leading-re** [reply]
I think the question instead is why Filament does not support basic Laravel functionality like model events -- and if it does, where they fit into Filament's lifecycle. Can this be documented?

**Comment by khanakia** [comment]
I am also looking for the same solution but could not find any.

@leandrocfe Filament hooks are not usable as I am using Graphql API, and Rest API too so I have some global validations rules that must always validate that's why I did add them within the booted function
